pyomo/opt/base/results.py

```diff
@@ -10,10 +10,10 @@
 
 __all__ = [ 'AbstractResultsReader', 'ReaderFactory' ]
 
-from pyomo.common.factory import Factory
+from pyomo.common.factory import CachedFactory
 
 
-ReaderFactory = Factory('problem reader')
+ReaderFactory = CachedFactory('problem reader')
```

Comment:
The reader object doesn't maintain any state, so we can cache the reader object.

Reply:
I don't think that is a safe assertion.  Just because the current reader implementation happens to not store state, doesn't mean that a future reader wouldn't store helper information on `self`.  This makes me nervous as it would be one of the few Factories that implements a Singleton interface (and it is exactly the singleton interface that caused memory leaks and heartburn in the past).

Reply:
While I don't agree with your tacit assertion that global data is evil, I've removed all use of CachedFactory in Pyomo.